buildSrc/src/main/java/org/elasticsearch/gradle/test/RestApiSpecForTestingPlugin.java

```diff
@@ -0,0 +1,141 @@
+package org.elasticsearch.gradle.test;
+
+import org.elasticsearch.gradle.VersionProperties;
+import org.elasticsearch.gradle.info.BuildParams;
+import org.elasticsearch.gradle.tool.Boilerplate;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.SourceSet;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+//TODO: change logger.info to logger.debug
+public class RestApiSpecForTestingPlugin implements Plugin<Project> {
+
+    private static final Logger logger = Logging.getLogger(RestApiSpecForTestingPlugin.class);
+    private static final String EXTENSION_NAME = "restApiSpec";
+    private static final String apiDir = "rest-api-spec/api";
+    private static final String testDir = "rest-api-spec/test";
+
+    @Override
+    public void apply(Project project) {
+
+        RestApiSpecExtension extension = project.getExtensions().create(EXTENSION_NAME, RestApiSpecExtension.class);
+        // need to defer to after evaluation to allow the custom extension to be populated
+        project.afterEvaluate(p -> {
+            try {
+                // copy tests
+                boolean hasCopiedTests = false;
+                if (extension.shouldCopyCoreTests()) {
```

Comment:
I think this may go without saying after your comment about lazy properties, but with your new direction instead we would _always_ create the copy tasks, they would simply be skipped in cases where we didn't configure any includes. In this case Gradle would report the task as `NO-SOURCE` I believe, indicating that task was scheduled for execution, but didn't have any work to perform.

Reply:
The latest update to the PR should address this.